tests/Controller/ProductVariantApiTest.php

```diff
@@ -0,0 +1,125 @@
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_product_variant_when_it_does_not_exist()
+    {
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('GET', sprintf('/api/v1/products/%s/variants/-1', $products['product1']->getId()), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_product_variant()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+        $product_variants = $this->loadFixturesFromFile('resources/product_variants.yml');
+
+        $this->client->request('GET', sprintf('/api/v1/products/%s/variants/%s', $products['product1']->getId(), $product_variants['productVariant2']->getId()), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'product_variant/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_product_variants()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+        $this->loadFixturesFromFile('resources/product_variants.yml');
+
+        $this->client->request('GET', sprintf('/api/v1/products/%s/variants/', $products['product1']->getId()), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'product_variant/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_paginating_the_index_of_product_variant()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+        $this->loadFixturesFromFile('resources/product_variants.yml');
+
+        $this->client->request('GET', sprintf('/api/v1/products/%s/variants/', $products['product1']->getId()), ['page' => 2], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'product_variant/paginated_index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_product_variant()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+
+        $data =
+<<<EOT
+        {
+            "code": "MONSTER_MUG",
+            "on_hand": 0,
```

Comment:
To create a product variant we should require only code. Rest should be optional